instrumentation/github.com/gocql/gocql/gocql_test.go

```diff
@@ -0,0 +1,479 @@
+	InstrumentWithProvider(controller.Provider())
+	return controller
+}
+
+// recordEqual checks that the given metric name and instrumentation names are equal.
+func recordEqual(t *testing.T, expected testRecord, actual export.Record) {
+	descriptor := actual.Descriptor()
+	assert.Equal(t, expected.Name, descriptor.Name())
+	assert.Equal(t, expected.MeterName, descriptor.InstrumentationName())
+	for _, label := range expected.Labels {
+		actualValue, ok := actual.Labels().Value(label.Key)
+		assert.True(t, ok)
+		assert.NotNil(t, actualValue)
+		// Can't test equality of host id
+		if label.Key != cassHostIDKey {
+			assert.Equal(t, label.Value, actualValue)
+		} else {
+			assert.NotEmpty(t, actualValue)
+		}
+	}
+}
+
+func numberEqual(t *testing.T, expected metric.Number, agg aggregation.Aggregation) {
+	kind := agg.Kind()
+	switch kind {
+	case aggregation.SumKind:
+		if sum, ok := agg.(aggregation.Sum); !ok {
+			t.Fatal("missing sum value")
+		} else {
+			if num, err := sum.Sum(); err == nil {
+				assert.Equal(t, expected, num)
+			} else {
+				t.Fatal("missing value")
+			}
+		}
+	case aggregation.ExactKind:
+		if mmsc, ok := agg.(aggregation.MinMaxSumCount); !ok {
+			t.Fatal("missing aggregation")
+		} else {
+			if max, err := mmsc.Max(); err == nil {
+				assert.Equal(t, expected, max)
+			} else {
+				t.Fatal("missing sum")
+			}
+		}
+	default:
+		t.Fatalf("unexpected kind %s", kind)
+	}
+}
+
+// beforeAll creates the testing keyspace and table if they do not already exist.
+func beforeAll() {
+	cluster := gocql.NewCluster("localhost")
+	cluster.Consistency = gocql.LocalQuorum
+	cluster.Keyspace = "system"
+
+	session, err := cluster.CreateSession()
+	if err != nil {
+		log.Fatalf("failed to connect to database during beforeAll, %v", err)
+	}
+
+	err = session.Query(
+		fmt.Sprintf(
+			"create keyspace if not exists %s with replication = { 'class' : 'SimpleStrategy', 'replication_factor' : 1 }",
+			keyspace,
+		),
+	).Exec()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	cluster.Keyspace = keyspace
+	session, err = cluster.CreateSession()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	err = session.Query(
+		fmt.Sprintf("create table if not exists %s(id UUID, title text, PRIMARY KEY(id))", tableName),
+	).Exec()
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+// afterEach truncates the table used for testing.
+func afterEach() {
+	cluster := gocql.NewCluster("localhost")
+	cluster.Consistency = gocql.LocalQuorum
+	cluster.Keyspace = keyspace
+	session, err := cluster.CreateSession()
+	if err != nil {
+		log.Fatalf("failed to connect to database during afterEach, %v", err)
+	}
+	if err = session.Query(fmt.Sprintf("truncate table %s", tableName)).Exec(); err != nil {
+		log.Fatalf("failed to truncate table, %v", err)
+	}
+}
+
+func TestMain(m *testing.M) {
+	if _, present := os.LookupEnv("INTEGRATION"); !present {
+		fmt.Println("--- SKIP: to enable integration test, set the INTEGRATION environment variable")
+		os.Exit(0)
+	}
```

Comment:
Do we need to update our CI system to be able to run these?

Reply:
I used the same pattern here as in the [mongo integration](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/2baa89862fad76c1ed79eb74f36c9978c2a7a512/instrumentation/go.mongodb.org/mongo-driver/mongo_test.go#L33-L40)

Just looking at the circle ci check, it doesn't appear to be running integration tests like this. This test relies on a running cassandra instance, and the mongo tests rely on mongo. Is there an interest in setting something like this up?

Reply:
I think we want to support this, but it is out of scope for this PR.

I opened https://github.com/open-telemetry/opentelemetry-go-contrib/issues/160 to track this work. If you're interested, we we would certainly be happy to have help on that :smile: .

Reply:
Would love to help!